src/core/mac/mac_frame.cpp

```diff
@@ -193,6 +199,34 @@ uint8_t Frame::FindDstPanIdIndex(void) const
     return index;
 }
 
+bool Frame::IsDstPanIdPresent(uint16_t aFcf) const
+{
+    bool present          = true;
+    bool isDstAddrPresent = IsDstAddrPresent(aFcf);
+
+    if ((aFcf & kFcfFrameVersionMask) == kFcfFrameVersion2015)
+    {
+        bool isPanIdCompressed = aFcf & kFcfPanidCompression;
+        bool isSrcAddrPresent  = IsSrcAddrPresent(aFcf);
+        // According to IEEE 802.15.4-2015 Table 7-2
+        if ((!isDstAddrPresent && !isSrcAddrPresent && !isPanIdCompressed) ||
+            (isDstAddrPresent && !isSrcAddrPresent && isPanIdCompressed) ||
+            (!isDstAddrPresent && isSrcAddrPresent && isPanIdCompressed) ||
+            (!isDstAddrPresent && isSrcAddrPresent && !isPanIdCompressed) ||
+            ((aFcf & kFcfDstAddrMask) == kFcfDstAddrExt && (aFcf & kFcfSrcAddrMask) == kFcfSrcAddrExt &&
+             isPanIdCompressed))
+        {
+            present = false;
+        }
+    }
```

Comment:
Why wrap it with `OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT`? It should be generic for OT1.2. I think wrap it with `OT_THREAD_VERSION_1_2` is sufficient. Thoughts?

Reply:
TIME_SYNC feature might be enabled independently with Thread 1.2 features, which also uses header ie.

Reply:
I prefer to use `OT_THREAD_VERSION_1_2`. Is TIME_SYNC related to dstPanId? For example, if we use a 1.1 device which enabled TIME_SYNC, when it parses a frame it should still not support 2015 frame, right?

Reply:
make sense to me.

Reply:
We have reached another consensus that we still need to use `OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT` because when we use `TIME_SYNC`, we would use version 2015 frame in 1.1, referring: #4836